src/app/services/validators/customValidators.ts

```diff
@@ -0,0 +1,17 @@
+import { AbstractControl, ValidatorFn } from '@angular/forms';
+
+export class CustomValidators {
+    static dateLessThan(dateField1: string, dateField2: string, validatorField: { [key: string]: boolean }): ValidatorFn {
+        return (c: AbstractControl): { [key: string]: boolean } | null => {
+            console.log(dateField1);
+            console.log(dateField2);
+            const d1 = new Date(c.get(dateField1).value).getTime();
+            const d2 = new Date(c.get(dateField2).value).getTime();
```

Comment:
It's minor but I tend to avoid encoding or shortening names even in small blocks of code. 

e.g. There's no nothing wrong with date1/date2 instead of d1/d2 and control instead of c.